src/Symfony/Component/HttpFoundation/HeaderUtils.php

```diff
@@ -143,6 +143,42 @@ public static function unquote(string $s): string
         return preg_replace('/\\\\(.)|"/', '$1', $s);
     }
 
+    /**
+     * Find the session header amongst the headers that are to be sent, remove it, and return
+     * it so the caller can process it further.
+     */
+    public static function popSessionCookie(string $sessionName, string $sessionId): ?string
```

Comment:
This method is really different from the other ones, as it does not deal with a header value, but with PHP headers themselves. So I would not expose it here as a public utility. I would keep it as `@internal` somewhere else (especially given than one of the 2 places using it will disappear when dropping support for PHP 7.2 and older)

Reply:
Good point. Any suggestion on where to keep it? Maybe introduce a  `Symfony\HttpFoundation\Session\SessionUtils` class?

Reply:
@stof I've created a new `Symfony\HttpFoundation\Session\SessionUtils` class, marked it as `internal` and moved it there. Does that seem better to you?